### PR TITLE
Add FIt-SNE

### PR DESCRIPTION
**Pull request by raivivek**
Signed-off-by: Vivek Rai <mail@raivivek.in>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/fit-sne/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#! /bin/sh
+
+g++ -std=c++11 -O3  src/sptree.cpp src/tsne.cpp src/nbodyfft.cpp -o fast_tsne \
```
Comment (scopatz) [inline recipes/fit-sne/build.sh]: Please use $CXX to get the right compiler path, rather than `g++`

File: recipes/fit-sne/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - fftw
+
+test:
+  commands:
+    - fast_tsne 2>&1 | grep 'Done'
+
+about:
+  home: https://github.com/KlugerLab/FIt-SNE
+  license: Multiple
+  license_file: LICENSE.txt
```
Comment (scopatz) [inline recipes/fit-sne/meta.yaml]: Also, you need a licence_family entry

Comment (raivivek) [reply recipes/fit-sne/meta.yaml]: @scopatz This particular package includes license from multiple sources -- what would be an appropriate value for for license_family and license in such a case? Thanks!

Comment (raivivek) [reply recipes/fit-sne/meta.yaml]: I used "OTHER" based on my review of conda-build source -- let me know if that's not appropriate.

File: recipes/fit-sne/bld.bat [hunk 3]
```diff
@@ -0,0 +1,4 @@
+move "%RECIPE_DIR%\FItSNE.exe" "%LIBRARY_BIN%"
```
Comment (scopatz) [inline recipes/fit-sne/bld.bat]: You need `if errorlevel 1 exit 1` below each command

File: recipes/fit-sne/meta.yaml (hunk 2 above)
Comment (scopatz) [reply recipes/fit-sne/meta.yaml]: That's fine

File: recipes/fit-sne/bld.bat (hunk 3 above)
Comment (raivivek) [reply recipes/fit-sne/bld.bat]: Thanks, I updated the code.

**Comment by raivivek** [comment]
@scopatz Looks like the Windows build is failing. For Windows, I am simply trying to copy a pre-built binary provided by the original repository, so it should not be too complicated...

**Comment by scopatz** [comment]
You should probably actually do a build on windows

**Comment by raivivek** [comment]
@scopatz I have skipped the Windows build for now due to two reasons: a) My lack of unfamiliarity with the build system to get it work properly, and b) the package is primarily intended to be used on Unix systems / bioconda distributed packages. Let me know if this looks good to merge.

File: recipes/fit-sne/build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#! /bin/sh
+
+export CPPFLAG="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
```
Comment (scopatz) [inline recipes/fit-sne/build.sh]: These should inherit from the flag in the environment:

```
export CPPFLAG="$CXXFLAGS -I$PREFIX/include"
export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
```

Comment (raivivek) [reply recipes/fit-sne/build.sh]: Okay, resolved in latest push.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**